php/commands/core.php

```diff
@@ -235,6 +259,29 @@ private static function _extract( $tarball, $dest ) {
 		self::_rmdir( dirname( $tempdir ) );
 	}
 
+	private static function _extract_zip( $zipfile, $dest ) {
+		if ( ! class_exists( 'ZipArchive' ) ) {
+			throw new \Exception( 'Extracting a zip file requires PharData' );
```

Comment:
@danielbachhuber It's caught here: https://github.com/stephenharris/wp-cli/blob/9eea025df9b69eab496b6c2a5dc3207a05702f4a/php/commands/core.php#L216

R.e. the early check, makes sense